aspnetcore/host-and-deploy/windows-service.md

```diff
@@ -5,110 +5,160 @@ description: Learn how to host an ASP.NET Core app in a Windows Service.
 
-  When creating a project in Visual Studio, use the **ASP.NET Core Application (.NET Framework)** template.
+1. Make the following changes in `Program.Main`:
 
-* If the app receives requests from the Internet (not just from an internal network), it must use the [HTTP.sys](xref:fundamentals/servers/httpsys) web server (formerly known as [WebListener](xref:fundamentals/servers/weblistener) for ASP.NET Core 1.x apps) rather than [Kestrel](xref:fundamentals/servers/kestrel). Use of IIS in a reverse proxy server configuration with Kestrel is an option for edge deployments. For more information, see [When to use Kestrel with a reverse proxy](xref:fundamentals/servers/kestrel#when-to-use-kestrel-with-a-reverse-proxy).
+   * Call [host.RunAsService](/dotnet/api/microsoft.aspnetcore.hosting.windowsservices.webhostwindowsserviceextensions.runasservice) instead of `host.Run`.
 
-## Get started
+   * If the code calls `UseContentRoot`, use a path to the app's published location instead of `Directory.GetCurrentDirectory()`.
+
+     ::: moniker range=">= aspnetcore-2.0"
+
+     [!code-csharp[](windows-service/sample/Program.cs?name=ServiceOnly&highlight=3-4,7,11)]
 
-This section explains the minimum changes required to set up an existing ASP.NET Core project to run in a service.
+     ::: moniker-end
 
-1. Install the NuGet package [Microsoft.AspNetCore.Hosting.WindowsServices](https://www.nuget.org/packages/Microsoft.AspNetCore.Hosting.WindowsServices/).
+     ::: moniker range="< aspnetcore-2.0"
 
-2. Make the following changes in `Program.Main`:
+     [!code-csharp[](windows-service/sample_snapshot/Program.cs?name=ServiceOnly&highlight=3-4,8,13)]
 
-   * Call `host.RunAsService` instead of `host.Run`.
+     ::: moniker-end
 
-   * If the code calls `UseContentRoot`, use a path to the publish location instead of `Directory.GetCurrentDirectory()`.
+1. Publish the app to a folder. Use [dotnet publish](/dotnet/articles/core/tools/dotnet-publish) or a [Visual Studio publish profile](xref:host-and-deploy/visual-studio-publish-profiles) that publishes to a folder.
 
-   # [ASP.NET Core 2.x](#tab/aspnetcore2x/)
+   To publish the sample app from the command line, run the following command in a console window from the project folder:
 
-   [!code-csharp[](windows-service/sample/Program.cs?name=ServiceOnly&highlight=3-4,7,12)]
+   ```console
+   dotnet publish --configuration Release --output c:\svc
+   ```
 
-   # [ASP.NET Core 1.x](#tab/aspnetcore1x/)
+1. Open a command shell with administrative privileges to use the [sc.exe](https://technet.microsoft.com/library/bb490995) command-line tool to create the service. Use the `sc create <SERVICE_NAME> binPath= "<PATH_TO_SERVICE_EXECUTABLE>"` command. The `binPath` value is the path to the app's executable, which includes the executable file name. **The space between the equal sign and the quote character that starts the path is required.**
```

Comment:
Open a command shell with administrative privileges. The command shell will .. clean this up - just split the sentence in two. --  to use the [sc.exe](https://technet.microsoft.com/library/bb490995) command-line tool to create the service.